client/client.go

```diff
@@ -254,13 +255,21 @@ func (cli *Client) Close() error {
 	return nil
 }
 
+// checkVersion manually triggers API version negotiation (if configured).
+// This allows for version-dependent code to use the same version as will
+// be negotiated when making the actual requests, and for which cases
+// we cannot do the negotiation lazily.
+func (cli *Client) checkVersion(ctx context.Context) {
+	if cli.negotiateVersion && !cli.negotiated.Load() {
+		cli.NegotiateAPIVersion(ctx)
```

Comment:
There was a PR from @corhere that fixes this by using a channel to store the version and the negotiation state: 
https://github.com/moby/moby/pull/43738

Which looks quite nice IMO, it seems that it went stale though.

Reply:
Yeah, I can probably remove the atomic.Bool

I recall the other PR; I know there were some concerns about the large amount of code impacted; which is why I was considering if it would make sense to use the context for the version (as I can see that the API version to use would always be scoped to the context (assuming here that 1 request means 1 context).

Reply:
Yeah I think removing it for now makes sense. Otherwise this could give a false impression that it actually somewhat solves the issue.

Reply:
Oh! Thanks for the ping; forgot to update. Did so now, so PTAL 👍